packages/contracts-bedrock/src/L2/TeaWAPOracle.sol

```diff
@@ -35,6 +40,51 @@ contract TeaWAPOracle {
     ///         setting the storage value to the fallback.
     uint256 public constant MAX_ORACLE_DOWNTIME = 5 minutes;
 
+    /// @notice SECURITY FIX: Minimum sane price for TEA/ETH ratio to prevent oracle manipulation
+    ///         Set to 100,000 TEA per ETH (100k * 1e18)
+    /// @dev REASON: Prevents gradual manipulation attacks where attacker slowly drains pool over
+    ///      many blocks to manipulate TWAP (not detectable by flash loan protection).
+    ///      Example attack: Drain pool from 2M → 200M TEA/ETH over 100 blocks, oracle would
+    ///      accept 100x price without bounds, causing users to pay 100x gas fees.
```

Comment:
After a single block, when the price is wrong, market makers would arb this back. What am I missing?